Packs/CrowdStrikeMalquery/Integrations/CrowdStrikeMalquery/CrowdStrikeMalquery.py

```diff
@@ -0,0 +1,425 @@
+import demistomock as demisto
+from CommonServerPython import *  # noqa: E402 lgtm [py/polluting-import]
+from CommonServerUserPython import *  # noqa: E402 lgtm [py/polluting-import]
+
+# Disable insecure warnings
+requests.packages.urllib3.disable_warnings()
+
+# CONSTANTS
+DATE_FORMAT = '%Y-%m-%dT%H:%M:%SZ'
+VENDOR_NAME = 'CrowdStrike Malquery'
+DBOT_SCORE = {
+    'unknown': 0,
+    'clean': 1,
+    'unwanted': 2,
+    'malware': 3,
+    'malicious': 3,
+}
+
+# Note: True life time of token is actually 30 mins
+TOKEN_LIFE_TIME = 28
+
+
+def get_passed_mins(start_time, end_time_str, tz=None):
+    """
+        Returns the time passed in mins
+        :param start_time: Start time in datetime
+        :param end_time_str: End time in str
+        :return: The passed mins in int
+    """
+    time_delta = start_time - datetime.fromtimestamp(end_time_str, tz)
+    return time_delta.seconds / 60
+
+
+def remove_None_values_keys(d):
+    return {
+        key: value
+        for key, value in d.items()
+        if value is not None
+    }
+
+
+class Client(BaseClient):
+    """
+    Client will implement the service API, and should not contain any Demisto logic.
+    Should only do requests and return data.
+    """
+
+    def __init__(self, base_url, verify, proxy, client_id, client_secret):
+        super().__init__(base_url=base_url, verify=verify, proxy=proxy)
+        self.client_id = client_id,
+        self.client_secret = client_secret
+
+    def http_request(self, *args, headers=None, **kwargs):
+        """
+        Overrides Base client request function, retrieves and adds to headers access token before sending the request.
+
+        Returns:
+            requests.Response: The http response
+        """
+        token = self.get_access_token()
+        default_headers = {
+            'Authorization': f'Bearer {token}',
+            'Content-Type': 'application/json',
+        }
+        if headers:
+            default_headers.update(headers)
+
+        return super()._http_request(*args, headers=default_headers, **kwargs)  # type: ignore[misc]
+
+    def get_access_token(self):
+        """
+           Obtains access and refresh token from server.
+           Access token is used and stored in the integration context until expiration time.
+           After expiration, new refresh token and access token are obtained and stored in the
+           integration context.
+
+           Returns:
+               str: Access token that will be added to authorization header.
+       """
+        now = datetime.now()
+        integration_context = demisto.getIntegrationContext()[0] \
```

Comment:
why would it be a list?

Reply:
@yaakovi - when I tested the integration using the demisto plugin on pycharm, it failed because the integration context was a list so I added the option.
should I remove it?